samza-core/src/main/java/org/apache/samza/operators/MessageStreamImpl.java

```diff
@@ -163,10 +163,11 @@ public K getOtherKey(M message) {
 
   @Override
   public <K> MessageStream<M> partitionBy(Function<M, K> parKeyExtractor) {
-    MessageStreamImpl<M> intStream = this.graph.createIntStream(parKeyExtractor);
+    int opId = graph.getNextOpId();
+    MessageStreamImpl<M> intStream = this.graph.generateIntStreamFromOpId(opId, parKeyExtractor);
     OutputStream<M> outputStream = this.graph.getOutputStream(intStream);
     this.registeredOperatorSpecs.add(OperatorSpecs.createPartitionOperatorSpec(outputStream.getSinkFunction(),
-        this.graph, outputStream));
+        this.graph, outputStream, opId));
```

Comment:
Issue: I think the `graph` parameter is only used to generate the operatorId later. If we're already getting the ID here here, should not pass both to avoid confusion.

We can also do a consistent pass over all operators separately, I'd prefer to get and pass the ID (like here) instead of the whole graph, unless @nickpan47 had some other usage of `graph` in mind.